metricbeat/mb/testing/modules.go

```diff
@@ -221,38 +220,51 @@ func NewPushMetricSetV2(t testing.TB, config interface{}) mb.PushMetricSetV2 {
 // capturingReporterV2 stores all the events and errors from a metricset's
 // Run method.
 type capturingReporterV2 struct {
-	Events []mb.Event
-	DoneC  chan struct{}
-	count  int32
+	events []mb.Event
+	doneC  chan struct{}
+	mutex  sync.Mutex
 }
 
 // Event stores the passed-in event into the events array
 func (r *capturingReporterV2) Event(event mb.Event) bool {
-	r.Events = append(r.Events, event)
-	atomic.AddInt32(&r.count, 1)
+	r.mutex.Lock()
+	r.events = append(r.events, event)
```

Comment:
Should we use `defer r.mutex.Unlock()` here to have the unlock close to the `mutex`? We have similar pattern in most of the code.

Reply:
I was a little inconsistent with the mutex unlock, doing it manually in some places and using defer when the return statement required the mutex to be kept in locked state. I was trying to avoid the `defer` overhead but that is stupid as this is just a testing helper.